### PR TITLE
block: Add helper to construct block from beacon payload

### PR DESCRIPTION
**Pull request by g11tech**
Add a handy static method to construct block from beacon payload which is more readily available via beacon apis: https://ethereum.github.io/beacon-APIs/#/Beacon/getBlockV2

### CODE REVIEW

**Comment by holgerd77** [comment]
Just labelled this on occasion, hope that's ok.

Interesting PR! 🙂

File: packages/block/src/from-beacon-payload.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  timestamp: string
+  extra_data: string
+  base_fee_per_gas: string
+  block_hash: string
+  transactions: string[]
+  withdrawals?: BeaconWithdrawal[]
+  excess_data_gas?: string
+}
+
+export function beaconToExecutionPayload(payload: BeaconPayload): ExecutionPayload {
```
Comment (acolytec3) [inline packages/block/src/from-beacon-payload.ts:33]: Can we rename this `beaconBlockToExecutionPayload` and add some docs to explain what it does and how to use it?

**Review by acolytec3 — CHANGES_REQUESTED** [review]
Code looks great.  Just need a little bit of additional docs in the new helper method and this should be ready to merge!

File: packages/block/src/from-beacon-payload.ts (hunk 1 above)
Comment (g11tech) [reply packages/block/src/from-beacon-payload.ts:33]: its not a beacon block, its execution payload in beacon block ( see the jq path here: `curl localhost:5052/eth/v2/beacon/blocks/56610 | jq .data.message.body.execution_payload`), let me know what you finally want to keep

Comment (acolytec3) [reply packages/block/src/from-beacon-payload.ts:33]: So maybe name it something like `executionPayloadFromBeaconBlock` and then add TypeDocs specifying what the input/outputs are.  The doc could be something like:
```
/**
 * Converts a beacon block execution payload JSON object to the {@link ExecutionPayload} data needed to construct a {@link Block}.
 * The JSON data can be retrieved from a consensus layer (CL) client on this Beacon API `/eth/v2/beacon/blocks/[block number]`
 */
 ```

Comment (g11tech) [reply packages/block/src/from-beacon-payload.ts]: applied changes

**Comment by holgerd77** [comment]
Oh. Something very fishy going on with `test-client`. 🙂

**Review by acolytec3 — APPROVED** [review]
Looks great, especially with the new docs!  Will merge once the CI passes.

**Comment by holgerd77** [comment]
Really nice PR! 🙂👍

**Pull request merged**